### DISCUSSION TITLE
useWatch + Field Array => Watch Multiple Fields

### DISCUSSION THREAD

**Discussion by rdick** [react-hook-form/react-hook-form#6820] [Q&A]
Hello, 

I have an ordering form that will use React hook forms AND FIELD ARRAY

I want to be able to monitor the following values on each line:

- Unit Price
- Quantity

So I can calculate TOTAL PRICE Per Line( eg. unit price X quantity = total price per line) . (see example below)

Then I want calculate OVERALL total price of ALL Lines. How would I do this without watching the WHOLE field array.

![image](https://user-images.githubusercontent.com/62073529/137502065-7c03b511-7812-42b3-b892-1db717d7db67.png)

**Comment by bluebill1049** [comment]
you can smartly watch by x colmn and y column to reduce the re-render and computation.

**Reply by rdick** [reply]
Hmmm, that isn't working for me

eg. if I watch x Row => and then set value of something in x Row => it re-renders infinitely (set value -> update -> set value etc...)

I have figured out how to get things in the Y column. => great idea!
